src/fheroes2/dialog/dialog_selectscenario.cpp

```diff
@@ -208,6 +226,15 @@ namespace
 
         text.draw( xCoordinate, yCoordinate, display );
     }
+
+    template <class ShowFunction>
+    void ShowIfFound( ScenarioListBox & listbox, const fheroes2::Point & mouseLocation, ShowFunction && function )
+    {
+        const Maps::FileInfo * item = listbox.GetFromPosition( mouseLocation );
+        if ( item ) {
+            std::forward<ShowFunction>( function )( item );
+        }
+    }
```

Comment:
In this case, there is no need to use a [universal reference](https://isocpp.org/blog/2012/11/universal-references-in-c11-scott-meyers), using it here does not give anything (that's why all these complaints about missing `std::forward`).

```suggestion
    template <class ShowFunction>
    void ShowIfFound( ScenarioListBox & listbox, const fheroes2::Point & mouseLocation, const ShowFunction & function )
    {
        const Maps::FileInfo * item = listbox.GetFromPosition( mouseLocation );
        if ( item ) {
            function( item );
        }
    }
```

Reply:
Done. Thanks for linked article, it really helps to know the difference between rvalue and universal references.